### PR TITLE
Add swc to documentation

### LINKED ISSUE #454: Documentation for non-babel compilers (SWC)
@pmmmwh, I see this pull request here https://github.com/pmmmwh/react-refresh-webpack-plugin/pull/450 

Is there anything missing for us to merge it? I know myself and a bunch of other folks would like to see it added :) 

(cc @Austaras)

### PR DESCRIPTION
**Pull request by Austaras**
Fixes #454

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +183,60 @@ module.exports = {
...
+### With swc-loader
+
+Your `@swc/core` version must be at least 1.2.52 and `swc-loader` version greater than 0.1.13 is recommened.
+
+> ⚠ `@swc/core` and `swc-loader` are maintained by the community not by Facebook.
+
+```js
+const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');
+const webpack = require('webpack');
+const ReactRefreshTypeScript = require('react-refresh-typescript');
```
Comment (fforres) [inline README.md:195]: Not an approval @Austaras 
But QQ. is `react-refresh-typescript` something that's needed for fast-refresh to work?

Comment (Austaras) [reply README.md]: My mistake. Fixed

Comment (fforres) [reply README.md]: Thanks :D

File: README.md [hunk 2]
```diff
@@ -183,6 +183,59 @@ module.exports = {
 };
 ```
 
+### With swc-loader
+
+Your `@swc/core` version must be at least 1.2.52 and `swc-loader` version greater than 0.1.13 is recommened.
```
Comment (pmmmwh) [inline README.md:188]:
```suggestion
Your `@swc/core` version must be at least 1.2.52 and `swc-loader` version greater than 0.1.13 is recommended.
```

File: examples/typescript-with-swc/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "using-typescript-without-babel",
```
Comment (pmmmwh) [inline examples/typescript-with-swc/package.json:2]:
```suggestion
  "name": "using-typescript-with-swc",
```

File: examples/typescript-with-swc/public/index.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>TS React App</title>
```
Comment (pmmmwh) [inline examples/typescript-with-swc/public/index.html:6]:
```suggestion
    <title>SWC React App</title>
```

File: examples/typescript-with-swc/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                parser: {
+                  syntax: 'typescript',
+                  tsx: true,
+                  dynamicImport: true,
+                },
+                transform: {
+                  react: {
+                    runtime: 'automatic',
+                    refresh: isDevelopment,
+                  },
```
Comment (pmmmwh) [inline examples/typescript-with-swc/webpack.config.js:32]: Do we need to set `development` here, or does SWC's automatic runtime handle that as well?

Comment (Austaras) [reply examples/typescript-with-swc/webpack.config.js:32]: See README L216, or should we copy the same comment here?

**Review by pmmmwh — APPROVED** [review]

**Comment by pmmmwh** [comment]
Thanks!

**Pull request merged**